Source/Csla/Configuration/Fluent/CslaDataPortalConfiguration.cs

```diff
@@ -31,13 +33,36 @@ public static CslaDataPortalConfiguration DataPortal(this ICslaConfiguration con
     /// <param name="builder"></param>
     public static CslaDataPortalConfiguration AddServerSideDataPortal(this CslaDataPortalConfiguration builder)
     {
+      return AddServerSideDataPortal(builder, null);
+    }
+
+    /// <summary>
+    /// Add services required to host the server-side
+    /// data portal.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="options"></param>
+    public static CslaDataPortalConfiguration AddServerSideDataPortal(this CslaDataPortalConfiguration builder, Action<DataPortalServerOptions> options)
+    {
+      var opt = new DataPortalServerOptions();
+      options?.Invoke(opt);
       var services = builder.Services;
-      services.TryAddTransient(typeof(Server.IDataPortalServer), typeof(Csla.Server.DataPortal));
-      services.TryAddTransient<Server.DataPortalSelector>();
-      services.TryAddTransient<Server.SimpleDataPortal>();
-      services.TryAddTransient<Server.FactoryDataPortal>();
-      services.TryAddTransient<Server.DataPortalBroker>();
-      services.TryAddSingleton(typeof(Server.Dashboard.IDashboard), typeof(Csla.Server.Dashboard.NullDashboard));
+      services.AddScoped((p) => opt);
+      services.AddScoped(typeof(IAuthorizeDataPortal), opt.AuthorizerProviderType);
+      services.AddScoped((p) => opt.InterceptorProviders);
+      if (opt.ObjectFactoryLoaderType != null)
+        services.AddScoped(typeof(IObjectFactoryLoader), opt.ObjectFactoryLoaderType);
+      if (opt.ActivatorType != null)
+        services.AddTransient(typeof(IDataPortalActivator), opt.ActivatorType);
+      if (opt.ExceptionInspectorType != null)
+        services.AddScoped(typeof(IDataPortalExceptionInspector), opt.ExceptionInspectorType);
```

Comment:
I wonder if the code might be simpler and less error-prone if there was always an exception inspector, by creating a NullExceptionInspector for scenarios where the less secure option of allowing exceptions to be propagated was required.